src/test/kotlin/org/wfanet/measurement/measurementconsumer/stats/VariancesTest.kt

```diff
@@ -3330,6 +3330,111 @@ class VariancesTest {
     }
   }
 
+  @Test
+  fun `computeMetricVariance returns for reach-frequency`() {
+    val vidSamplingIntervalWidth = 1e-4
+    val totalReach = 1L
+    val reachDpParams = DpParams(0.05, 1e-15)
+    val reachMeasurementParams =
+      ReachMeasurementParams(
+        VidSamplingInterval(0.0, vidSamplingIntervalWidth),
+        reachDpParams,
+        NoiseMechanism.GAUSSIAN,
+      )
+    val reachMeasurementVarianceParams =
+      ReachMeasurementVarianceParams(totalReach, reachMeasurementParams)
+    val reachMeasurementVariance =
+      VariancesImpl.computeMeasurementVariance(
+        DeterministicMethodology,
+        reachMeasurementVarianceParams,
+      )
+
+    val maximumFrequency = 5
+    val relativeFrequencyDistribution =
+      (1..maximumFrequency).associateWith { (maximumFrequency - it) / 10.0 }
+    val frequencyDpParams = DpParams(0.2, 1e-15)
+    val frequencyMeasurementParams =
+      FrequencyMeasurementParams(
+        VidSamplingInterval(0.0, vidSamplingIntervalWidth),
+        frequencyDpParams,
+        NoiseMechanism.GAUSSIAN,
+        maximumFrequency,
+      )
+    val frequencyMeasurementVarianceParams =
+      FrequencyMeasurementVarianceParams(
+        totalReach,
+        reachMeasurementVariance,
+        relativeFrequencyDistribution,
+        frequencyMeasurementParams,
+      )
+
+    val weight = 2
+    val coefficient = weight * weight.toDouble()
+
+    val weightedFrequencyMeasurementVarianceParams =
+      WeightedFrequencyMeasurementVarianceParams(
+        binaryRepresentation = 1,
+        weight = weight,
+        measurementVarianceParams = frequencyMeasurementVarianceParams,
+        methodology = DeterministicMethodology,
+      )
+
+    val (rKVars, rKPlusVars, nKVars, nKPlusVars) =
+      VariancesImpl.computeMetricVariance(
+        FrequencyMetricVarianceParams(listOf(weightedFrequencyMeasurementVarianceParams))
+      )
+
+    val expectedRK =
+      listOf(130523240799.76, 110944754739.79, 104418592319.84, 110944753539.91, 130523238400.0)
+        .map { it * coefficient }
+    val expectedRKPlus =
+      listOf(0.0, 130523240799.75995, 215363345459.78998, 215363344259.90997, 130523238400.0).map {
+        it * coefficient
+      }
+    val expectedNK =
+      listOf(
+          2.5828737279268425e+23,
+          2.195442669924104e+23,
+          2.06629897600801e+23,
+          2.1954426461785614e+23,
+          2.582873680435757e+23,
+        )
+        .map { it * coefficient }
+    val expectedNKPlus =
+      listOf(
+          1978861168399.0,
+          2.5828737279307992e+23,
+          4.261741614272709e+23,
+          4.2617415905271664e+23,
+          2.582873680435757e+23,
+        )
+        .map { it * coefficient }
+
```

Comment:
I'm trying to understand the number of the four lists in line 3388--3409. Where do these expected values come from?